src/h5web/providers/hsds/hsds-api.ts

```diff
@@ -182,33 +187,42 @@ export class HsdsApi extends ProviderApi {
     id: HsdsId,
     path: string,
     name: string,
-    depth: number
+    isChild = false
   ): Promise<HsdsEntity<Group>> {
     const { attributeCount, linkCount } = await this.fetchGroup(id);
 
     // Fetch attributes and links in parallel
-    // If recursion depth has been reached, don't fetch links at all
     const [attributes, links] = await Promise.all([
       attributeCount > 0
         ? this.fetchAttributes('groups', id)
         : Promise.resolve([]),
-      linkCount > 0 && depth > 0 ? this.fetchLinks(id) : Promise.resolve([]),
+      !isChild
+        ? linkCount > 0
```

Comment:
I would have kept the ternary with `!isChild && linkCount > 0`...

Reply:
But then, I'd have to check `isChild` again in the `false` branch, to choose between `[]` and `undefined`.